### PR TITLE
Display 'kubernetes' clouds as 'k8s'

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

When listing or showing clouds, instead of "kubernetes", print "k8s".
Also a couple of drive by fixes: new make target to update microk8s operator image and a tweak to k8s api version reported in hook context.

## QA steps

juju clouds
juju clouds --format yaml
juju show-cloud foo

## Bug reference

https://bugs.launchpad.net/juju/+bug/1814670

### CODE REVIEW

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -303,7 +303,10 @@ func (k *kubernetesClient) APIVersion() (string, error) {
 	if err != nil {
 		return "", errors.Annotatef(err, "got '%s' querying API version", string(body))
 	}
-	return fmt.Sprintf("%s.%s", info.Major, info.Minor), nil
+	version := info.GitVersion
+	// git version is "vX.Y.Z", strip the "v"
+	version = strings.Trim(version, "v")
+	return version, nil
```
Comment (Veebers) [inline caas/kubernetes/provider/k8s.go]: So this has changed from returning "x.y" to now returning "x.y.z"? There was no test update needed?

Comment (wallyworld) [reply caas/kubernetes/provider/k8s.go]: There's a test that calling APIVersion() invokes the correct ReST URL but to weave in the return result would have required an extra 100s of lines of mocks and/or stubs seeing as the underlying code deals with a rest.Request struct containing an actual embedded http client.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**